From e5065166ecb975b9255805fe53d83f1dc2f2682a Mon Sep 17 00:00:00 2001
From: Lorenzo Bianconi <lorenzo@kernel.org>
Date: Sun, 6 Mar 2022 20:34:13 +0100
Subject: [PATCH 8/9] DFS: Enable CSA for background radar detection

Rely on hostapd_dfs_request_channel_switch() to enable CSA for
background radar detection switching back to the selected channel.

Signed-off-by: Lorenzo Bianconi <lorenzo@kernel.org>
Signed-off-by: Manish Dharanenthiran <quic_mdharane@quicinc.com>
---
 src/ap/dfs.c | 10 +++++++---
 1 file changed, 7 insertions(+), 3 deletions(-)

diff --git a/src/ap/dfs.c b/src/ap/dfs.c
index eb91e9a..82b2aa2 100644
--- a/src/ap/dfs.c
+++ b/src/ap/dfs.c
@@ -1198,6 +1198,8 @@ hostapd_dfs_is_background_event(struct hostapd_iface *iface, int freq)
 static int
 hostapd_dfs_start_channel_switch_background(struct hostapd_iface *iface)
 {
+	u8 current_vht_oper_chwidth = hostapd_get_oper_chwidth(iface->conf);
+
 	iface->conf->channel = iface->radar_background.channel;
 	iface->freq = iface->radar_background.freq;
 	iface->conf->secondary_channel =
@@ -1208,10 +1210,12 @@ hostapd_dfs_start_channel_switch_background(struct hostapd_iface *iface)
 		iface->conf, iface->radar_background.centr_freq_seg1_idx);
 
 	hostpad_dfs_update_background_chain(iface);
-	hostapd_disable_iface(iface);
-	hostapd_enable_iface(iface);
 
-	return 0;
+	return hostapd_dfs_request_channel_switch(
+		iface, iface->conf->channel, iface->freq,
+		iface->conf->secondary_channel, current_vht_oper_chwidth,
+		hostapd_get_oper_centr_freq_seg0_idx(iface->conf),
+		hostapd_get_oper_centr_freq_seg1_idx(iface->conf));
 }
 
 
-- 
2.17.1

